source/common/http/utility.cc

```diff
@@ -547,10 +547,14 @@ void Utility::sendLocalReply(const bool& is_reset, const EncodeFunctions& encode
   if (local_reply_data.is_grpc_) {
     response_headers->setStatus(std::to_string(enumToInt(Code::OK)));
     response_headers->setReferenceContentType(Headers::get().ContentTypeValues.Grpc);
-    response_headers->setGrpcStatus(
-        std::to_string(enumToInt(local_reply_data.grpc_status_
-                                     ? local_reply_data.grpc_status_.value()
-                                     : Grpc::Utility::httpToGrpcStatus(enumToInt(response_code)))));
+
+    if (response_headers->getGrpcStatusValue().empty()) {
```

Comment:
please add the behavioral change to the current release notes.

Do we have any Envoy filters which would have their status code affected as a result?  If not I'm inclined to consider this a minor bug fix and not require runtime guards but ccing @mattklein123 for a second opinion

Reply:
ping on this one!

Reply:
This should be fine.  It consists with the way of how other fields, e.g. body_text, are handled:  only set a header if  it is empty.    Normally, sending local_reply needs to set necessary fields for the protocol.  but if they are already set, on the corner cases of local_redirect, or header_to_add, then not to set it. 

So this is minor bug fix.

Reply:
Sounds good, please add it to the release notes as a bug fix then!

Reply:
added release note